python/res/enkf/res_config.py

```diff
@@ -55,18 +70,68 @@ class ResConfig(BaseCClass):
 
 
     def __init__(self, user_config_file=None, config=None, throw_on_error=True):
+
         self._errors, self._failed_keys = None, None
         self._assert_input(user_config_file, config, throw_on_error)
 
         if config is not None:
             config_content = self._build_config_content(config)
+            config_dir = config_content.getValue(ConfigKeys.CONFIG_DIRECTORY)
+
+            subst_config = SubstConfig(config_content=config_content)
+            site_config = SiteConfig(config_content=config_content)
+            rng_config = RNGConfig(config_content=config_content)
+            analysis_config = AnalysisConfig(config_content=config_content)
+            plot_config = PlotSettings(config_content=config_content)
+            ecl_config = EclConfig(config_content=config_content)
+            log_config = LogConfig(config_content=config_content)
+
+            ert_workflow_list = ErtWorkflowList(ert_workflow_list=subst_config.subst_list,
+                                                config_content=config_content)
+
+            hook_manager = HookManager(workflow_list=ert_workflow_list,
+                                       config_content=config_content)
+
+            ert_templates = ErtTemplates(parent_subst=subst_config.subst_list,
+                                         config_content=config_content)
+
+            ensemble_config = EnsembleConfig(config_content=config_content,
+                                             grid=ecl_config.getGrid(),
+                                             refcase=ecl_config.getRefcase())
+
+            model_config = ModelConfig(config_content=config_content,
+                                       data_root=config_dir,
+                                       joblist=site_config.get_installed_jobs(),
+                                       last_history_restart=ecl_config.getLastHistoryRestart(),
+                                       sched_file=ecl_config._get_sched_file(),
+                                       refcase=ecl_config.getRefcase())
+
+            configs = [
```

Comment:
?

Reply:
Later there is a need to iterate over all configs and convert to C reference.